cpp/unit/action.cpp

```diff
@@ -79,6 +96,130 @@ void UnitAction::move_to(Unit &target, bool use_range) {
 	this->entity->invoke(cmd);
 }
 
+TargetAction::TargetAction(Unit *u, graphic_type gt, UnitReference r, coord::phys_t rad)
+	:
+	UnitAction(u, gt),
+	target{r},
+	repath_attempts{10},
+	end_action{false},
+	radius{rad} {
+	this->update_distance();
+}
+
+TargetAction::TargetAction(Unit *u, graphic_type gt, UnitReference r)
+	:
+	TargetAction(u, gt, r, adjacent_range(u)) {
+}
+
+void TargetAction::update(unsigned int time) {
```

Comment:
According to `UnitAction`'s docs, "each action has its own update functionality".

Please summarize the kind of update functionality that TargetAction has (either in the class doc, the doc for `TargetAction::update`, or in a comment block here in the impl).